### PR TITLE
Cmr 7183 - Update UMM-C UseConstraints LicenseUrl to LicenseURL to be consistent with the other models and updated Description to be a field instead of an object.

### PR DESCRIPTION
**Pull request by eereiter**

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj [hunk 1]
```diff
@@ -393,3 +393,24 @@
...
+(defmethod interface/migrate-umm-version [:collection "1.16.1" "1.16.2"]
+  [context c & _]
+  ;; Can't use the same solution as used in 1.16.2 -> 1.16.1 using update because when :UseConstraints
+  ;; is nil, it will be turned into (:Description nil :LIcenseUrl nil :LicenseText nil) which will
+  ;; fail validation. This is the same issue going from 1.9 -> 1.10 above.
+  (let [use-constraints (:UseConstraints c)
+        use-constraints (-> use-constraints
+                            (assoc :Description (get-in use-constraints [:Description :Description]))
+                            (set/rename-keys {:LicenseUrl :LicenseURL})
+                            util/remove-nils-empty-maps-seqs)]
```
Comment (mschmele) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj:406]:
```suggestion
  (let [use-constraints (-> c
                            :UseConstraints
                            (assoc :Description (get-in use-constraints [:Description :Description]))
                            (set/rename-keys {:LicenseUrl :LicenseURL})
                            util/remove-nils-empty-maps-seqs)]
```

File: umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj [hunk 2]
```diff
@@ -393,3 +393,24 @@
...
+     (if use-constraints
+       (assoc c :UseConstraints use-constraints)
+       (dissoc c :UseConstraints))))
+
+(defmethod interface/migrate-umm-version [:collection "1.16.2" "1.16.1"]
+  [context c & _]
+  (update c :UseConstraints #(-> %
+                                 (assoc :Description {:Description (get % :Description)})
+                                 (set/rename-keys {:LicenseURL :LicenseUrl})
+                                 util/remove-nils-empty-maps-seqs)))
```
Comment (mschmele) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj:416]: Use the long-form `fn` notation here

https://clojuredocs.org/clojure.core/fn

File: umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj (hunk 1 above)
Comment (eereiter) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj:406]: I need the use-constraints variable in the (get-in function on line 404 for the assoc to work correctly.

Comment (eereiter) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj]: I have modified the code.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj (hunk 2 above)
Comment (eereiter) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj]: I have changed this to use the long form.

**Review by mschmele — APPROVED** [review]

**Review by jceaser — CHANGES_REQUESTED** [review]
The Ingest Documentation has not been updated to indicate the latest UMM version that can be used. At minimum, the line "application/vnd.nasa.cmr.umm+json;version=1.14" should be changed in the ingest API.

**Comment by eereiter** [comment]
I updated the ingest documentation version examples to the latest version.

**Review by jceaser — APPROVED** [review]

**Pull request merged**